### ISSUE TITLE
Release Candidate 2021.01-RC5 Hi-DPI Linux Text Issue

### ISSUE THREAD

**Issue by mholiv** [openscad/openscad#3614]
In Release Candidate 2021.01-RC5 the text appears blurry and looks bad overall on Linux with Hi-DPI displays. 

This is regression from OpenSCAD version 2019.05 where things look good.

My UI Relevant Details:
```
X version number:    11.0
X.Org version: 1.20.10
Kernel: 5.4.85-1-MANJARO
Screen: dimensions: 7843x2160 pixels (2073x571 millimeters) (2 4K Displays)
Global Scale: 137.5%
```
My System:
```
Operating System: Manjaro Linux
KDE Plasma Version: 5.20.4
KDE Frameworks Version: 5.77.0
Qt Version: 5.15.2
Kernel Version: 5.4.85-1-MANJARO
OS Type: 64-bit
Processors: 12 × AMD Ryzen 5 1600X Six-Core Processor
Memory: 31.4 GiB of RAM
Graphics Processor: Radeon RX 580 Series
```

![new_vs_old](https://user-images.githubusercontent.com/16260356/104768630-d6d94580-5722-11eb-9e37-622f6433ceff.png)

**Comment by rcolyer** [comment]
You have a lot more pixels to work with there than I do, but I currently don't see any difference at a pixel to pixel level between the fonts between the releases then and now.  Could you also add two text files with the contents of Help / Library Info for both versions on your system?  The top and the long sections at the bottom (e.g. font path) might be important.

**Comment by t-paul** [comment]
I don't see that issue on my system (Dell XPS 13 with 4K built-in display 3840x2160 @ 200% scale + Full HD external display 1920x1080 @ 100% scale).

![grafik](https://user-images.githubusercontent.com/1330241/104816550-497a1c00-581c-11eb-8a9c-8c15d17c8139.png)

Maybe Qt/QScintilla don't support fractional scale?

**Comment by mholiv** [comment]
@t-paul I tested it at 100% 200% and 125%. Yah it looks like non integer scaling is the root cause here.

The strange part is that in version 2019.05 things where find and looked good.

In both the release candidate and the 2019.05 build the non-text-editor UI scales really well. It's just the text editor that has the issues. This leads me to believe it is a QScintilla issue.

I found a thread about a similar issue here.
https://github.com/flathub/org.octave.Octave/issues/13

I guess the root question is, "Why can 2019.05 handle Hi-DPI Non integer scaling but RC5 can't?" Is is a pretty noticeable regression for me. :(

**Comment by rcolyer** [comment]
@mholiv Are you by any chance able to do a compile and test of the PR above that contains a potential fix?  So far none of us are able to reproduce this issue, so we're not confident it actually fixes anything.  While I don't think disabling the buffering as done there causes any serious issues, I'd be hesitant to merge it without knowing that it actually fixes at least one problem.

**Comment by mholiv** [comment]
@rcolyer Just compiled and can confirm that your PR works. Text looks great on HI-DPI displays using non-integer scaling. 

Thanks for helping with this.

Side note When compiling the dependencies building the build script I ran into an error while compiling glib2. I ran the dependency check script and it said everything was good so I moved forward. Just FYI.

**Comment by t-paul** [comment]
Nice! I'm not seeing a difference on my system, so crossing fingers it's not causing issues on other platforms. I'll add the patch to the release.

**Comment by t-paul** [comment]
Fixed via #3620.
